node/impl/full/chain.go

```diff
@@ -589,6 +593,111 @@ func (m *ChainModule) ChainGetMessage(ctx context.Context, mc cid.Cid) (*types.M
 	return cm.VMMessage(), nil
 }
 
+func (a ChainAPI) ChainExportRangeInternal(ctx context.Context, head, tail types.TipSetKey, cfg api.ChainExportConfig) error {
+	headTs, err := a.Chain.GetTipSetFromKey(ctx, head)
+	if err != nil {
+		return xerrors.Errorf("loading tipset %s: %w", head, err)
+	}
+	tailTs, err := a.Chain.GetTipSetFromKey(ctx, tail)
+	if err != nil {
+		return xerrors.Errorf("loading tipset %s: %w", tail, err)
+	}
+	fileName := fmt.Sprintf("./snapshot_%d_%d_%d.car", tailTs.Height(), headTs.Height(), time.Now().Unix())
```

Comment:
This arguably should be written to the repo path, which you can get by adding `repo.LockedRepo` to the ChainAPI struct, and calling `.Path` on it.

Reply:
`ERROR: initializing node: applying node options failed: repo is already locked (lotus daemon already running)`

Reply:
```diff
diff --git a/node/impl/full/chain.go b/node/impl/full/chain.go
index 5dedc6991..6923e311c 100644
--- a/node/impl/full/chain.go
+++ b/node/impl/full/chain.go
@@ -42,6 +42,7 @@ import (
        "github.com/filecoin-project/lotus/lib/oldpath"
        "github.com/filecoin-project/lotus/lib/oldpath/oldresolver"
        "github.com/filecoin-project/lotus/node/modules/dtypes"
+       "github.com/filecoin-project/lotus/node/repo"
 )

 var log = logging.Logger("fullnode")
@@ -93,6 +94,8 @@ type ChainAPI struct {

        // BaseBlockstore is the underlying blockstore
        BaseBlockstore dtypes.BaseBlockstore
+
+       Repo repo.LockedRepo
 }

 func (m *ChainModule) ChainNotify(ctx context.Context) (<-chan []*api.HeadChange, error) {
@@ -606,7 +609,7 @@ func (a ChainAPI) ChainExportRangeInternal(ctx context.Context, head, tail types
                return xerrors.Errorf("Height of head-tipset (%d) must be greater or equal to the height of the tail-tipset (%d)", headTs.Height(), tailTs.Height())
        }

-       fileName := fmt.Sprintf("./snapshot_%d_%d_%d.car", tailTs.Height(), headTs.Height(), time.Now().Unix())
+       fileName := filepath.Join(a.Repo.Path(), fmt.Sprintf("snapshot_%d_%d_%d.car", tailTs.Height(), headTs.Height(), time.Now().Unix()))
        absFileName, err := filepath.Abs(fileName)
        if err != nil {
                return err
```